editor/store/test/selectors.js

```diff
@@ -2616,224 +2639,335 @@ describe( 'selectors', () => {
 				},
 				currentPost: {},
+				preferences: {
+					insertUsage: {},
+				},
+				blockListSettings: {},
+				settings: {},
 			};
-
-			expect( getFrecentInserterItems( state, true, 3 ) ).toMatchObject( [
-				{ name: 'core/block', initialAttributes: { ref: 123 } },
-				{ name: 'core/image', initialAttributes: {} },
-				{ name: 'core/paragraph', initialAttributes: {} },
-			] );
+			const items = getInserterItems( state );
+			const testBlockBItem = items.find( ( item ) => item.id === 'core/test-block-b' );
+			expect( testBlockBItem.isDisabled ).toBe( true );
 		} );
 
-		it( 'should weight by time', () => {
+		it( 'should give common blocks a medium utility', () => {
 			const state = {
-				preferences: {
-					insertUsage: {
-						'core/image': { time: Date.now() - 1000, count: 2, insert: { name: 'core/image' } },
-						'core/paragraph': { time: Date.now() - 4000, count: 3, insert: { name: 'core/paragraph' } },
-					},
-				},
 				editor: {
 					present: {
-						blockOrder: [],
+						blocksByUID: {},
+						blockOrder: {},
+						edits: {},
 					},
 				},
 				sharedBlocks: {
 					data: {},
 				},
+				currentPost: {},
+				preferences: {
+					insertUsage: {},
+				},
+				blockListSettings: {},
+				settings: {},
 			};
-
-			expect( getFrecentInserterItems( state, true, 2 ) ).toMatchObject( [
-				{ name: 'core/image', initialAttributes: {} },
-				{ name: 'core/paragraph', initialAttributes: {} },
-			] );
+			const items = getInserterItems( state );
+			const testBlockBItem = items.find( ( item ) => item.id === 'core/test-block-b' );
+			expect( testBlockBItem.utility ).toBe( 1 );
 		} );
 
-		it( 'should be backwards-compatible with old preferences values', () => {
+		it( 'should give used blocks a high utility and set a frecency', () => {
 			const state = {
-				preferences: {
-					insertUsage: {
-						'core/image': { time: Date.now(), count: 1, insert: { name: 'core/image' } },
-						'core/paragraph': { time: undefined, count: 5, insert: { name: 'core/paragraph' } },
-					},
-				},
 				editor: {
 					present: {
-						blockOrder: [],
+						blocksByUID: {},
+						blockOrder: {},
+						edits: {},
 					},
 				},
 				sharedBlocks: {
 					data: {},
 				},
-			};
-
-			expect( getFrecentInserterItems( state, true, 2 ) ).toMatchObject( [
-				{ name: 'core/paragraph', initialAttributes: {} },
-				{ name: 'core/image', initialAttributes: {} },
-			] );
-		} );
-
-		it( 'should pad list out with blocks from the common category', () => {
-			const state = {
+				currentPost: {},
 				preferences: {
 					insertUsage: {
-						'core/image': { time: 1000, count: 2, insert: { name: 'core/paragraph' } },
+						'core/test-block-b': { count: 10, time: 1000 },
 					},
 				},
+				blockListSettings: {},
+				settings: {},
+			};
+			const items = getInserterItems( state );
+			const sharedBlock2Item = items.find( ( item ) => item.id === 'core/test-block-b' );
+			expect( sharedBlock2Item.utility ).toBe( 2 );
```

Comment:
Value `2` is hard to follow in here, the description says `high`.

Reply:
👍 added some constants in 6efa3afd2